unit-tests/live/syncer/test-throughput.cpp

```diff
@@ -0,0 +1,248 @@
+            {
+                std::lock_guard<std::mutex> lock(_mutex);
+                auto stream_type = std::string(f.get_profile().stream_name());
+                // Only IR fps is relevant for this test, IR1 and IR2 have same fps so it is enough to get only one of them
+                if (!(stream_type == "Infrared 1" || stream_type == "Color"))
+                    return;
+                if (!f.supports_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS) || !f.supports_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP))
+                    return;
+                auto frame_num = f.get_frame_number();
+                auto actual_fps = (float)f.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS);
+                auto frame_arrival = f.get_frame_metadata(RS2_FRAME_METADATA_FRAME_TIMESTAMP); // usec
+                if (stream_type == "Infrared 1")
+                {
+                    _ir_frames_arrival_info.push_back({ (float)frame_arrival, actual_fps });
+                    if (actual_fps != prev_fps) // exposure new value is finally set
+                    {
+                        auto t_now = std::chrono::system_clock::now();
+                        _delta = (float)std::chrono::duration_cast<std::chrono::seconds>(t_now - t_start).count();
+                    }
+                    prev_fps = actual_fps;
+                }
+                if (stream_type == "Color")
+                    _rgb_frames_arrival_info.push_back({ (float)frame_arrival, actual_fps });
+            };
+
+            float delta = (float)std::chrono::duration_cast<std::chrono::seconds>(t_end - t_start).count();
+            while (delta < RECEIVE_FRAMES_TIME)
+            {
+                if (delta > RECEIVE_FRAMES_TIME/2)
+                    set_ir_exposure(); // set exposure to 18000 after 5 sec
+
+                auto fs = _sync.wait_for_frames();
+                for (const rs2::frame& ff : fs)
+                {
+                    process_frame(ff);
+                }
+                t_end = std::chrono::system_clock::now();
+                delta = (float)std::chrono::duration_cast<std::chrono::seconds>(t_end - t_start).count();
+            }
+        }  
+        void analysis(configuration test, std::vector <std::pair<float, float>>& frames_arrival_info, std::string stream_type = "Infrared 1")
+        {
+            if (frames_arrival_info.empty())
+                return;
+            float fps = 0.0f;
+            while (fps < 10) // skip unstable frames at the beginning
+            {
+                fps = frames_arrival_info.front().second;
+                frames_arrival_info.erase(frames_arrival_info.begin());
+            }
+            std::vector<std::pair<float, float>>  vec[2];
+            for (auto& frm : frames_arrival_info)
+            { 
+                auto fps_ratio = frm.second / fps; // used ratio instead of precised fps because syncer doesn't change fps to exact value when running RGB
```

Comment:
Check fps != 0
Use meaningful names for the vectors to understand the division. Alternatively write a comment